app/main.py

```diff
@@ -1,2 +1,81 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self._capacity = 8
+        self._load_factor = 2 / 3
+        self._size = 0
+        self._storage = [None] * self._capacity
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        idx = hash(key) % self._capacity
+        while True:
+            if self._storage[idx] is None:
+                self._storage[idx] = [key, hash(key), value]
+                self._size += 1
+                break
+            if (
+                key == self._storage[idx][0]
+                and hash(key) == self._storage[idx][1]
+            ):
+                self._storage[idx][2] = value
+                break
+            idx = (idx + 1) % self._capacity
+        if self._size >= self._capacity * self._load_factor:
+            self.resize()
+
+    def __getitem__(self, key: Any) -> None:
+        idx = hash(key) % self._capacity
+        end_idx = idx - 1
+        while True:
+            if (
+                self._storage[idx] is not None
+                and key == self._storage[idx][0]
+                and hash(key) == self._storage[idx][1]
+            ):
+                return self._storage[idx][2]
+            if idx == end_idx:
+                raise KeyError
+            idx = (idx + 1) % self._capacity
+
+    def __delitem__(self, key: Any) -> None:
+        for idx in range(self._capacity):
+            if self._storage[idx] is not None and self._storage[idx][0] == key:
+                self._storage[idx] = None
+                break
+        else:
+            raise KeyError(key)
+
+    def __len__(self) -> int:
+        return self._size
+
+    def __iter__(self) -> None:
+        for item in self._storage:
+            if item is not None:
+                yield item[2]
```

Comment:
It's a Dict, why we return only value? This shoudl return jkeys if values() or items() not specified